src/tiled/objectselectiontool.cpp

```diff
@@ -1431,20 +1433,45 @@ void ObjectSelectionTool::saveSelectionState()
 void ObjectSelectionTool::refreshCursor()
 {
     Qt::CursorShape cursorShape = Qt::ArrowCursor;
-
+    const MapObject *prevObject = nullptr;
     switch (mAction) {
     case NoAction: {
         const bool hasSelection = !mapScene()->selectedObjectItems().isEmpty();
 
         if ((mHoveredObjectItem || ((mModifiers & Qt::AltModifier) && hasSelection && !mHoveredHandle)) &&
                 !(mModifiers & Qt::ShiftModifier)) {
             cursorShape = Qt::SizeAllCursor;
+            MapObject *currentObject = mHoveredObjectItem->mapObject();
+
+            if (currentObject != prevObject) {
```

Comment:
`prevObject` is always going to still be `nullptr` here. `prevObject` will go out of scope after it is being assigned, so all the values assigned to it will never be used.